src/haystack/filter/parser.rs

```diff
@@ -17,6 +17,12 @@ pub struct Parser<Lexer> {
     lexer: Lexer,
 }
 
+// Internal constant tokens to be used when comparing against current tokens
+lazy_static! {

```

Comment:
A micro-optimization, basically avoid the minimal cost of constructing these during parsing.